dapr_agents/agents/durable.py

```diff
@@ -407,40 +614,100 @@ def call_llm(
         user_copy = dict(user_message) if user_message else None
         self._process_user_message(instance_id, task, user_copy)
 
-        if user_copy is not None:
+        # Only print user message when task is provided (turn 1)
+        # On subsequent turns, user message is already in history and shouldn't be reprinted
+        if user_copy is not None and task is not None:
             self.text_formatter.print_message({str(k): v for k, v in user_copy.items()})
 
+        generate_kwargs = {
+            "messages": messages,
+        }
+        tools = self.get_llm_tools()
+        if tools:
+            generate_kwargs["tools"] = tools
+        if tools and self.execution.tool_choice is not None:
+            generate_kwargs["tool_choice"] = self.execution.tool_choice
+        if self.output_type:
+            generate_kwargs["response_format"] = self.output_type
+
         try:
-            response: LLMChatResponse = self.llm.generate(
-                messages=messages,
-                tools=self.get_llm_tools(),
-                tool_choice=self.execution_config.tool_choice,
-            )
+            llm_result = self.llm.generate(**generate_kwargs)
```

Comment:
is there a way we can avoid updating the func from defined/concrete params to accepting kwargs? That can be difficult to know what exactly to pass in without looking concretely at generate() implementation. I know its likely driven by tool_choice param and not wanting to set it if None, but can this be done within generate() itself instead of here pls?